### ISSUE TITLE
cilium stuck "Waiting on node-init" after upgrade from 1.10 to 1.11

### ISSUE THREAD

**Issue by dosmanak** [cilium/cilium#20211]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

/tmp/cilium-bootstrap.d/cilium-bootstrap-time is not created
The STARTUP_SCRIPT is run on worker node using nsentry. 
It does check the existence of checkpoint path defined in the script as
```
CHECKPOINT_PATH="${CHECKPOINT_PATH:-/tmp/startup-script.kubernetes.io_$(md5sum <<<"${STARTUP_SCRIPT}" | cut -c-32)}"
```
But previous instalation exported variable to "/tmp/node-init.cilium.io" in https://github.com/cilium/cilium/blob/02a0d13e006540124fe0f5fb6c8a9e4a77c4b563/install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml
and so the init pod does not ever run `do_startup_script` and so it does not create `/tmp/cilium-bootstrap.d/cilium-bootstrap-time`

### Cilium Version

1.10 -> 1.11.5

### Kernel Version

Linux aks-agentpool-10481151-vmss000002 5.4.0-1078-azure #81~18.04.1-Ubuntu SMP Mon Apr 25 23:16:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

1.22

### Anything else?

The initial installation was made using cilium-cli but then reinstalled using helm accoring to the documentation
at https://docs.cilium.io/en/stable/gettingstarted/k8s-install-helm/

```
helm install cilium cilium/cilium \
--version 1.11.5   \
--namespace kube-system   \
--set azure.enabled=true   \
--set azure.resourceGroup=$AZURE_NODE_RESOURCE_GROUP   \
--set azure.subscriptionID=$AZURE_SUBSCRIPTION_ID   \
--set azure.tenantID=$AZURE_TENANT_ID   \
--set azure.clientID=$AZURE_CLIENT_ID   \
--set azure.clientSecret=$AZURE_CLIENT_SECRET   \
--set tunnel=disabled   \
--set ipam.mode=azure   \
--set enableIPv4Masquerade=false   \
--set nodeinit.enabled=true \
--set hubble.ui.enabled=true \
--set hubble.relay.enabled=true
```

running manualy in cilium-node-init  on each worker node did unblock the cilium init containers.
```
nsenter -t 1 -m -u -i -n -p -- bash -c "${STARTUP_SCRIPT}"
```
the syntax were extracted from `/usr/bin/manage-startup-script.sh`

Restart of the nodeworker would also resolve the issue.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by christarazi** [comment]
Did you upgrade from the latest v1.10.x to v1.11.5?

**Comment by dosmanak** [comment]
> Did you upgrade from the latest v1.10.x to v1.11.5?

Well it was first shot install of cilium so I did not make it properly. I suppose I installed it using cilium cli (v0.10.7), so the first version was v0.10.11? Then I found in my bash history `helm install --version 1.11.1`. But It worked in the meantime so I am lost...
Now the cilium seems to be running but it drops packet even without any policy defined. I will start from scratch.
